packages/backend/src/apps/paysg/actions/create-payment-form-submission/schema.ts

```diff
@@ -0,0 +1,87 @@
+import emailValidator from 'email-validator'
+import { z } from 'zod'
+
+import { FORM_ID_LENGTH } from '@/apps/formsg/common/constants'
+
+import { normalizeSpecialChars } from '../create-payment/normalize-special-chars'
+
+const PAYMENT_FORM_LINK_REGEX = /^https:\/\/(staging.)?pay.gov.sg\/forms\/(.*)$/
+
+export const requestSchema = z.object({
+  formId: z
+    .string()
+    .trim()
+    .min(1, { message: 'Specify a form link' })
+    .transform((value, context) => {
+      const match = value.match(PAYMENT_FORM_LINK_REGEX)
+
+      if (!match || !match[0]) {
+        context.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: 'Invalid payment form link',
+        })
+        return z.NEVER
+      }
+
+      return match[2]
+    }),
+  formsg_form_id: z
+    .string()
+    .trim()
+    .length(FORM_ID_LENGTH, { message: 'Specify a valid FormSG form ID' }),
+  formsg_submission_id: z
+    .string()
+    .trim()
+    .min(1, { message: 'Specify a submission ID' }),
+  nonce: z.string().trim().min(1, { message: 'Specify a nonce field' }),
+  payer_name: z
```

Comment:
```suggestion
  nonce: z
    .string()
    .trim()
    .min(1, { message: 'Specify a FormSG reference field' }),
```

suggest to update the error message, not sure if users will understand what a nonce field is

Reply:
done in [9cf3cac](https://github.com/opengovsg/plumber/pull/870/commits/9cf3cacb42b3e9221d11c6e307b27b7a748042b5)